incremental_analysis/analysis_patcher.py

```
@@ -0,0 +1,544 @@
+            return apply_scope_patch(analysis, patch_scope, scope_patch)
+        except ValidationError as exc:
+            logger.warning(
+                "Scope patch generation failed for %s (attempt %d/%d): %s",
+                patch_scope.scope_id or "root",
+                attempt,
+                _PATCH_MAX_ATTEMPTS,
+                exc,
+            )
+            if raw_response is not None:
+                messages.append(_build_validation_feedback(raw_response, exc))
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)